api/client.go

```diff
@@ -657,19 +676,47 @@ func (c *Client) Token() string {
 
 // SetToken sets the token directly. This won't perform any auth
 // verification, it simply sets the token properly for future requests.
+// Setting a token will kill any routine polling a sink for a token.
 func (c *Client) SetToken(v string) {
-	c.modifyLock.Lock()
-	defer c.modifyLock.Unlock()
-
-	c.token = v
+	switch {
+	// no op
+	case v == "" && c.config.TokenFileSinkPath == "":
+
+	// a token is provided and should cancel polling
+	case v != "":
+		if c.pollingChannel != nil {
+			c.pollingChannel <- 1
```

Comment:
I would simply close the channel here, rather than reading or writing it.

Reply:
yes - that would be more straightforward